### PR TITLE
Blocker Slf4j Logger Backer

### PR DESCRIPTION
**Pull request by diogocanut**
Solves #440 .

### CODE REVIEW

File: slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala
Comment (danicheg) [inline slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala:63]: MiMa is arguing against this change https://github.com/typelevel/log4cats/actions/runs/5270003952/jobs/9948431964?pr=762#step:13:29
But since this is a package private class actually, we can workaround this warning like it's done here https://github.com/typelevel/log4cats/blob/main/core/shared/src/main/scala/org/typelevel/log4cats/PagingSelfAwareStructuredLogger.scala#L80

**Review by danicheg — COMMENTED** [review]
Sorry for slowpoking on reviewing this actually nice PR!

File: slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala
Comment (diogocanut) [reply slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala:63]: Pushed some changes, let me know if it's make sense

**Comment by danicheg** [comment]
Thanks for iterating over this. LGTM! But let's wait for a while if other maintainers want to put their two cents on this.

File: slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala
Comment (armanbilge) [inline slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala:77]: I wonder if these ones should be kept as `F.delay(...)`.

1. I understand that actually logging may be blocking, but I don't think I heard of a logger where checking the configuration is blocking.

2. I guess a client may check these to avoid doing some expensive work if logging is disabled. But if they are treated as blocking then checking these can be very expensive.

**Comment by danicheg** [comment]
Hey @diogocanut! Sorry for pinging, but your PR is good and I'd like to merge it, but there is a reasonable suggest from @armanbilge. Could you glance at it? Otherwise, I can do a final polish if you don't mind. Thank you!

File: slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala
Comment (diogocanut) [reply slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala]: Changed this to use `F.delay`. Thanks.

File: slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala
Comment (armanbilge) [inline slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala:44]: Sorry, I think this is another one that should stay as `delay`. Interacting with the `MDC` should not be blocking, it's just a thread-local mutable map.

Comment (diogocanut) [reply slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala]: Done. Let me know if there is something else.

File: slf4j/src/main/scala/org/typelevel/log4cats/slf4j/Slf4jLogger.scala
Comment (armanbilge) [inline slf4j/src/main/scala/org/typelevel/log4cats/slf4j/Slf4jLogger.scala:54]: Sorry 😂 can this one be `delay` as well? `getLoggerFromBlockingSlf4j` is just calling `new` right? To make an instance of our own class. Making a new instance of a class should not be blocking right?

Comment (diogocanut) [reply slf4j/src/main/scala/org/typelevel/log4cats/slf4j/Slf4jLogger.scala]: Done xd

**Review by armanbilge — APPROVED** [review]
thanks!

**Comment by danicheg** [comment]
Great work on collaborating and making a valuable contribution 👍🏻

**Pull request merged**
